### ISSUE TITLE
.pipe() after gulp.dest() throws ENOENT randomly

### ISSUE THREAD

**Issue by macu** [gulpjs/gulp#349]
Hi, I'm getting the same error around half the time I run this sort of task:

``` JavaScript
gulp.task('example', function() {
    return gulp.src('src/**/*.js')
        .pipe(gulp.dest('build'))
        .pipe(print())
        // ...
```

It doesn't seem to matter what the next step is (I've tried gulp-print, gulp-concat, gulp-uglify), or whether there are any steps before `gulp.dest`. It only seems to matter that there is a next step. The error is always about the first file in the first subdirectory, alphabetically matched by the source glob.

```
stream.js:94
      throw er; // Unhandled stream error in pipe.
            ^
Error: ENOENT, open '/home/macu/JSBuildDemo/build/app/a.js'
```

**Comment by briedis** [comment]
But what does it mean ""clean" task was running simultaneously"??

I'm getting the same error and can't solve it.

**Comment by dinoboff** [comment]
A task named `clean` deleting `build/` was probably running in parallel to that `example` task filling that same `build/` directory. He probably forgot to make that `clean` task async.

**Comment by macu** [comment]
Yes, I don't recall exactly now, but the actual task that was running, say "default", listed dependencies like ["clean", "example"]. Back in the day, I believe these dependency tasks were run in parallel.

**Comment by briedis** [comment]
Ok, I guess I narrowed it down.

I am using laravels elixir as a wrapper, and I was calling elixir.versioning, which registers a watcher. It really seems that it is async and that's why it fails.

So yeah, the problem is with another task running :)

Guess I just have to find out how to versioning synchronously.

**Comment by bjorn-ali-goransson** [comment]
For the record. The following is the remedy for a common cause of this error:

http://stackoverflow.com/a/32609091/3638471
